### PR TITLE
WIP Managed Async Publisher

### PR DESCRIPTION
**Pull request by scottf**
Define the managed async publish functionality

### CODE REVIEW

File: adr/ADR-43.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+| long holdPauseTime                          | the amount of time in ms to pause between checks when hold is on. Defaults to 100ms                                                                                                                                                                                          |
+| long waitTimeout                            | the timeout when waiting for a publish to be acknowledged. Defaults to 5000ms                                                                                                                                                                                                |
+| PublisherListener publisherListener         | a callback for the user to see what's going on in the workflow, see description of Flight later                                                                                                                                                                              |
+
+
+## PublisherListener Interface
+
+The callback interface for the user to get information about the publish workflow
```
Comment (Jarema) [inline adr/ADR-43.md:35]: The callback interface would not work in many languages as s good API.
This should be described in more general terms.

File: adr/ADR-43.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+|-----------------------------------------------------|--------------------------------------------------------------------------------------|
+| long getPublishTime()                               | the time when the actual js publish was executed. Used for waitTimeout determination |
+| CompletableFuture<PublishAck> getPublishAckFuture() | the future for the publish ack                                                       |
+| String getId()                                      | `<idPrefix>-<number>` for example                                                    |
+| String getSubject()                                 | the original message subject                                                         |
+| Headers getHeaders()                                | the original message headers                                                         |
+| byte[] getBody()                                    | the original message body                                                            |
+| PublishOptions getOptions()                         | the original message publish options (i.e. expectations)                             |
+
+## Behavior
```
Comment (Jarema) [inline adr/ADR-43.md:58]: The whole behavior description is very Java-specific and does not apply for the most clients.

**Review by Jarema — COMMENTED** [review]
@scottf I think that what you described is a very Java-specific.
Additionaly, I think it would be easier if any details about publishers would be part of the JetStream spec ADR, not a separate one.

File: adr/ADR-43.md (hunk 1 above)
Comment (scottf) [reply adr/ADR-43.md:35]: Such as?

**Pull request closed without merging**